### PR TITLE
[Backport 18.03] Circle CI migration to 2.0

### PR DESCRIPTION
**Pull request by fcrisciani**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.10
+FROM golang:1.10.2 as dev
```
[the remark this answers is not in the document]
Comment (fcrisciani) [reply Dockerfile]: I think I will put 1.10 again, anyway we are going to bump all the minor releases, wdyt?

[the remark this answers is not in the document]
Comment (selansen) [reply Dockerfile]: I think we should keep whatever moby points currently.

**Comment by selansen** [comment]
LGTM

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (fcrisciani) [reply Dockerfile]: within 2 patch releases? that sounds weird

[the remark this answers is not in the document]
Comment (thaJeztah) [reply Dockerfile]: Not sure if it was in the last changes, but it's not unprecedented

**Comment by fcrisciani** [comment]
will follow up with the golang update with the cherry-pick from master

**Pull request merged**